pkg/p2p/p2p.go

```diff
@@ -49,12 +49,17 @@ type Halter interface {
 type PickyNotifier interface {
 	Picker
 	Notifier
+	ReachabilityTracker
 }
 
 type Picker interface {
 	Pick(Peer) bool
 }
 
+type ReachabilityTracker interface {
```

Comment:
nitpicking, but usually single-method interfaces are named like `<function-name>-er`so `Pick()` -> `Picker`. it also helps to find more condensed names :) so maybe `ReachabilityUpdater`?